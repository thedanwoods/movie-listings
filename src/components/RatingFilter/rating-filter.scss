.rating-filter {
  padding: 0 0 16px;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 400;
  font-size: 13px;
}

.rating-filter__heading {
  font-weight: 700;
  padding: 0 0 4px 4px;
  font-size: 13px;
}

.rating-filter__select {
  appearance: none;
  -webkit-appearance: none;
  background-color: #2a1a1f;
  color: #bbb;
  border: none;
  cursor: pointer;
  width: 100%;
  padding: 4px;

  &:hover {
    background-color: #3a242b
  };

}

.rating-filter__option {
  appearance: none;
  -webkit-appearance: none;
}