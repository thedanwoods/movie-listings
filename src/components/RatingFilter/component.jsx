import React from 'react';
import PropTypes from 'prop-types';

import './rating-filter.scss';

// Create sequence 0 to 10 in 0.5 intervals
const sequence = Array.from(new Array(21).keys()).map(el => el / 2);

const displayRatingNumber = n =>
  `${n !== 0.5 ? Math.floor(n) : ''}${n === Math.floor(n) ? '' : '\u00BD'}`;

const RatingFilter = ({ current, onChange }) => (
  <div className="rating-filter">
    <h2 className="rating-filter__heading">Minimum rating</h2>
    <select
      id="rating-filter-dropdown"
      className="rating-filter__select"
      value={current}
      name="rating"
      onChange={e => onChange(e.target.value)}
    >
      {sequence.map(rating => (
        <option className="rating-filter__option" key={rating} value={rating}>
          {displayRatingNumber(rating)}
        </option>
      ))}
    </select>
  </div>
);

RatingFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  current: PropTypes.number.isRequired,
};

export default RatingFilter;
