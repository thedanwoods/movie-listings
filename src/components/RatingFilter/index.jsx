import { connect } from 'react-redux';
import RatingFilter from './component';
import ratingClicked from '../../actions/ratingClicked';

export const mapStateToProps = state => ({ current: state.ratingFilter });

export const mapDispatchToProps = dispatch => ({
  onChange: rating => {
    dispatch(ratingClicked(rating));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(RatingFilter);
