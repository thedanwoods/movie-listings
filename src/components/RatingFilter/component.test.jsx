import React from 'react';
import { shallow } from 'enzyme';
import RatingFilter from './component';

const mockChangeHandler = jest.fn();

const defaultProps = {
  current: 5.5,
  onChange: mockChangeHandler,
};

describe('<RatingFilter /> component', () => {
  it('renders without errors', () => {
    const ratingFilter = shallow(<RatingFilter {...defaultProps} />);

    expect(() => ratingFilter).not.toThrow();
    expect(ratingFilter.debug()).toMatchSnapshot();
  });

  it('calls onChange event handler on select change', () => {
    const ratingFilter = shallow(<RatingFilter {...defaultProps} />);
    expect(mockChangeHandler).not.toHaveBeenCalled();
    ratingFilter.find('#rating-filter-dropdown').simulate('change', { target: { value: '1' } });
    expect(mockChangeHandler).toHaveBeenCalled();
  })
});
