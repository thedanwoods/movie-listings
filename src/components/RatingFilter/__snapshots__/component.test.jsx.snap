// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingFilter /> component renders without errors 1`] = `
"<div className=\\"rating-filter\\">
  <h2 className=\\"rating-filter__heading\\">
    Minimum rating
  </h2>
  <select id=\\"rating-filter-dropdown\\" className=\\"rating-filter__select\\" value={5.5} name=\\"rating\\" onChange={[Function: onChange]}>
    <option className=\\"rating-filter__option\\" value={0}>
      0
    </option>
    <option className=\\"rating-filter__option\\" value={0.5}>
      ½
    </option>
    <option className=\\"rating-filter__option\\" value={1}>
      1
    </option>
    <option className=\\"rating-filter__option\\" value={1.5}>
      1½
    </option>
    <option className=\\"rating-filter__option\\" value={2}>
      2
    </option>
    <option className=\\"rating-filter__option\\" value={2.5}>
      2½
    </option>
    <option className=\\"rating-filter__option\\" value={3}>
      3
    </option>
    <option className=\\"rating-filter__option\\" value={3.5}>
      3½
    </option>
    <option className=\\"rating-filter__option\\" value={4}>
      4
    </option>
    <option className=\\"rating-filter__option\\" value={4.5}>
      4½
    </option>
    <option className=\\"rating-filter__option\\" value={5}>
      5
    </option>
    <option className=\\"rating-filter__option\\" value={5.5}>
      5½
    </option>
    <option className=\\"rating-filter__option\\" value={6}>
      6
    </option>
    <option className=\\"rating-filter__option\\" value={6.5}>
      6½
    </option>
    <option className=\\"rating-filter__option\\" value={7}>
      7
    </option>
    <option className=\\"rating-filter__option\\" value={7.5}>
      7½
    </option>
    <option className=\\"rating-filter__option\\" value={8}>
      8
    </option>
    <option className=\\"rating-filter__option\\" value={8.5}>
      8½
    </option>
    <option className=\\"rating-filter__option\\" value={9}>
      9
    </option>
    <option className=\\"rating-filter__option\\" value={9.5}>
      9½
    </option>
    <option className=\\"rating-filter__option\\" value={10}>
      10
    </option>
  </select>
</div>"
`;
